File: GraduateStudyWebApp/ExamLibSpa_UT/Pages/TS4GP/PubFun/stuTopPara.ts
﻿import { Dictionary } from "./tzDictionary.js";

export class stuTopPara {

    /// <summary>
    /// 顶部记录数
    /// </summary>    
    public TopSize: number = 0;

    /// <summary>
    /// 条件串
    /// </summary>
    public WhereCond: string = "";
    /// <summary>
    /// 排序表达式
    /// </summary>
    public OrderBy: string = "";

    /// <summary>
    /// 排除的检查字符串列表
    /// </summary>
    public Exclude: string = "";

    /// <summary>
    /// 是否检查SQL攻击
    /// </summary>
    public IsCheckSQLAttack: string = "false";

    public static GetMapParam(objTopPara: stuTopPara): Dictionary {
        //List < string > arrLst = new List<string>(objTopPara.Exclude);
        var mapParam: Dictionary = new Dictionary();
        mapParam.add("strWhereCond", objTopPara.WhereCond);
        mapParam.add("strOrderBy", objTopPara.OrderBy);
        mapParam.add("intTopSize", objTopPara.TopSize.toString());
        mapParam.add("arrExclude", objTopPara.Exclude);
        mapParam.add("bolIsCheckSQLAttack", objTopPara.IsCheckSQLAttack.toString());
        return mapParam;
        //var mapParam: Dictionary = new Dictionary()
        //{
        //    "strWhereCond" = objTopPara.WhereCond,
        //        "strOrderBy" = objTopPara.OrderBy,
        //        "intTopSize" = objTopPara.TopSize.toString(),
        //        ["arrExclude"] = clsArray.GetSqlInStrByArray(arrLst, false),
        //        ["bolIsCheckSQLAttack"] = objTopPara.IsCheckSQLAttack.ToString(),
        //    };
        //return dictParam;
    }



}

File: TzPubFunction/com/taishsoft/common/stuTopPara.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Runtime.Serialization;
using System.Web;

namespace com.taishsoft.common
{
    /// <summary>
    /// 获取顶部对象列表的参数对象-Static
    /// </summary>
    public static class stuTopPara_Static
    {
        /// <summary>
        /// 根据获取顶部对象列表的参数对象获取相关的字典参数
        /// </summary>
        /// <param name="objTopPara">获取顶部对象列表的参数对象</param>
        /// <returns>获取的相关字典参数</returns>
        public static Dictionary<string, string> GetDictParam(this stuTopPara objTopPara)
        {
            //List<string> arrLst = new List<string>(objTopPara.exclude);
           
            Dictionary<string, string> dictParam = new Dictionary<string, string>()
            {
                ["strWhereCond"] = objTopPara.whereCond,
                ["strOrderBy"] = objTopPara.orderBy,
                ["intTopSize"] = objTopPara.topSize.ToString(),
                ["arrExclude"] = objTopPara.exclude,
                ["bolIsCheckSQLAttack"] = objTopPara.isCheckSQLAttack.ToString(),
            };
            return dictParam;
        }
    }
    /// <summary>
    /// 获取顶部对象列表的参数对象
    /// </summary>
    [Serializable]
    [DataContractAttribute]
    public class stuTopPara
    {
      
        /// <summary>
        /// 顶部记录数
        /// </summary>
        [DataMember]//非常重要
        public int topSize { get; set; }

        /// <summary>
        /// 条件串
        /// </summary>
        [DataMember]//非常重要
        public string whereCond { get; set; }
        /// <summary>
        /// 排序表达式
        /// </summary>
        [DataMember]//非常重要
        public string orderBy { get; set; }

        /// <summary>
        /// 排除的检查字符串列表
        /// </summary>
        [DataMember]//非常重要
        public string exclude { get; set; }

        /// <summary>
        /// 是否检查SQL攻击
        /// </summary>
        [DataMember]//非常重要
        public string isCheckSQLAttack { get; set; }

    }
}